src/index.ts

```diff
@@ -296,6 +307,8 @@ export class Upload extends Pumpify {
     if (typeof cfg.generation === 'number') {
       cacheKeyElements.push(`${cfg.generation}`);
     }
+    // TODO: add another element for chunk size? Any conflicts possible here?
```

Comment:
The only situation I can imagine needing a second key would be if the a user started a non-chunk size upload that fails and then starts a chunk size upload before resuming the first. I don't see this is a very likely scenario.